static/app/components/organizations/projectPageFilter.tsx

```diff
@@ -0,0 +1,493 @@
+            />
+            <StyledBookmarkStar
+              project={project}
+              organization={organization}
+              visible={isFocused}
+              onToggle={(isBookmarked: boolean) => {
+                analytics('projectselector.bookmark_toggle', {
+                  org_id: parseInt(organization.id, 10),
+                  bookmarked: isBookmarked,
+                });
+              }}
+            />
+          </Fragment>
+        ),
+      };
+    };
+
+    const lastSelected = mapURLValueToNormalValue(pageFilterValue);
+    const listSort = (project: Project) => [
+      !lastSelected.includes(parseInt(project.id, 10)),
+      !project.isBookmarked,
+      project.slug,
+    ];
+
+    return nonMemberProjects.length > 0
+      ? [
+          {
+            key: 'my-projects',
+            label: t('My Projects'),
+            options: sortBy(memberProjects, listSort).map(getProjectItem),
+            showToggleAllButton: allowMultiple,
+          },
+          {
+            key: 'no-membership-header',
+            label:
+              memberProjects.length > 0 ? t('Other') : t("Projects I Don't Belong To"),
+            options: sortBy(nonMemberProjects, listSort).map(getProjectItem),
+            showToggleAllButton: allowMultiple && memberProjects.length > 0,
+          },
+        ]
+      : sortBy(memberProjects, listSort).map(getProjectItem);
+  }, [
+    organization,
+    allowMultiple,
+    memberProjects,
+    nonMemberProjects,
+    mapURLValueToNormalValue,
+    pageFilterValue,
+  ]);
+
+  const trigger = useCallback<NonNullable<MultipleSelectProps<number>['trigger']>>(
+    props => {
+      const isMemberProjectsSelected = memberProjects.every(p =>
+        value.includes(parseInt(p.id, 10))
+      );
+
+      const isNonMemberProjectsSelected = nonMemberProjects.every(p =>
+        value.includes(parseInt(p.id, 10))
+      );
+
+      const isMyProjectsSelected = isMemberProjectsSelected && memberProjects.length > 0;
+
+      const isAllProjectsSelected =
+        value.length === 0 || (isMyProjectsSelected && isNonMemberProjectsSelected);
+
+      const selectedProjects = value
+        .slice(0, 2) // we only need to know about the first two projects
+        .map(val =>
+          [...memberProjects, ...nonMemberProjects].find(
+            p => String(p.id) === String(val)
+          )
+        )
+        .filter((p): p is Project => !!p);
+
+      // Show 2 projects only if the combined string does not exceed maxTitleLength.
+      // Otherwise show only 1 project.
+      const projectsToShow =
+        selectedProjects[0]?.slug?.length + selectedProjects[1]?.slug?.length <= 23
+          ? selectedProjects.slice(0, 2)
+          : selectedProjects.slice(0, 1);
+
+      const label = isAllProjectsSelected
+        ? t('All Projects')
+        : isMyProjectsSelected
+        ? t('My Projects')
+        : projectsToShow.map(proj => trimSlug(proj.slug, 25)).join(', ');
```

Comment:
nit: 23 and 25 above here could be named variables so they have documented meaning

Reply:
Done!